packages/report-flaky-tests/lib/markdown.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * External dependencies
+ */
+import path from 'path';
+import { formatResultsErrors } from 'jest-message-util';
+import * as core from '@actions/core';
+
+/**
+ * Internal dependencies
+ */
+import { stripAnsi } from './strip-ansi';
+import { TEST_RESULTS_LIST, TEST_RESULT, metaData } from './constants';
+import type { MetaData, FlakyTestResult } from './types';
+
+type ParsedTestResult = {
```

Comment:
We generally have been putting all type definitions into `./types`.

Reply:
Not sure about this, I don't think we need to put every type in `./types` if it's only used in one file.

Reply:
It's the convention that we've been using in `@wordpress/components` which is where most of the project's TypeScript is currently. I think the motivation is to make it so that there is less "TypeScript noise" to read while reading the code.

Reply:
I don't think that's _noise_ though. Types should also be seen as documentation, just like JSDoc. If it's noise, then we should probably do the same with inline JSDoc and move most of them to a file like `jsdoc.js`. 😅

Another disadvantage of moving standalone types to `types.ts` is that we no longer know where the types are imported at first glance. It creates a mental overhead of not being able to relate the types with the file together.

I don't feel strongly about this either though, so if anyone thinks it should be consistent then I'd be happy to move it :).

Reply:
>  If it's noise, then we should probably do the same with inline JSDoc and move most of them to a file like jsdoc.js. 😅

We sometimes do in a way, when we do the `/** @typedef Blah import( './Blah' ) */` thing 😀

But all good, I don't feel strongly about this.